app/scripts/metamask-controller.js

```diff
@@ -3373,6 +3376,35 @@ export default class MetamaskController extends EventEmitter {
       );
     }
 
+    const tokenDetailsStandardIsERC1155 = isEqualCaseInsensitive(
+      details.standard,
+      TokenStandard.ERC1155,
+    );
+
+    if (tokenDetailsStandardIsERC1155) {
+      try {
+        const balance = await fetchERC1155Balance(
```

Comment:
It could be convenient to have `getTokenStandardAndDetails => ERC1155Standard.getDetails` perform the balance fetch, so it's already available in `details` for the ERC1155 case.  Especially since `details` already has a `balance` field where it could go.  But don't have to do this now.